estimagic/optimization/process_constraints.py

```diff
@@ -79,9 +79,15 @@ def process_constraints(constraints, params):
         warnings.filterwarnings(
             "ignore", message="indexing past lexsort depth may impact performance."
         )
+        params = params.copy()
         pc = _apply_constraint_killers(constraints)
         check_types(pc)
+        # selectors have to be processed before anything else happens to the params
         pc = _process_selectors(pc, params)
+        if "lower" not in params.columns:
```

Comment:
Looks like the same operation as in optimize. Maybe one is redundant or should be refactored to function.

Reply:
Done